src/m365/spfx/commands/project/project-github-workflow-model.ts

```diff
@@ -0,0 +1,21 @@
+export interface gitHubWorkflow {
+  name: string,
+  on: {
+    push: {
+      branches: string[];
+    },
+    "workflow_dispatch": any
+  },
+  jobs: {
+    "build-and-deploy": {
+      "runs-on": string,
+      steps: gitHubWorkflowStep[]
+    }
+  }
+}
+
+export interface gitHubWorkflowStep {
+  name: string,
+  uses: string,
+  with: any
```

Comment:
As we've got a finite set of properties that we're referencing in code, let's define all of them here to avoid typos. We can mark all of them as optional since we're not always using all of them

Reply:
yes now I understand. I hope.
done ✅